src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,643 @@
+		}
+
+		protected override void OnLoaded()
+		{
+			base.OnLoaded();
+			RefreshShape();
+		}
+
+		private CALayer _shapeLayer;
+
+		#region Measure / Arrange should be shared using Geometry instead of CGPath
+		private protected Size MeasureRelativeShape(Size availableSize)
+		{
+			var stretch = Stretch;
+			var userMinSize = new Size(MinWidth, MinHeight);
+			var userSize = new Size(Width, Height);
+
+			var size = userSize;
+
+			// If no user size defined on a given axis, we try to stretch along this axis
+			if (double.IsNaN(size.Width))
+			{
+				size.Width = stretch == Stretch.UniformToFill || HorizontalAlignment == HorizontalAlignment.Stretch
+					? availableSize.Width
+					: 0;
+			}
+			if (double.IsNaN(size.Height))
+			{
+				size.Height = stretch == Stretch.UniformToFill || VerticalAlignment == VerticalAlignment.Stretch
+					? availableSize.Height
+					: 0;
+			}
+
+			// In case userSize was not defined, we still have to apply the min size
+			size = size
+				.AtLeast(userMinSize)
+				.NumberOrDefault(userMinSize);
+
+			if (double.IsInfinity(size.Width) || double.IsInfinity(size.Height))
+			{
+				// The size is invalid (the userSize was not defined and we were not able to stretch), just hide the shape.
+				// Note: This will be overriden by the Layouter that will enforce the MinSize
+				return default;
+			}
+			else
+			{
+				return size;
+			}
+		}
+		private protected (Size shapeSize, Rect pathArea) ArrangeRelativeShape(Size finalSize)
+		{
+			var horizontal = HorizontalAlignment;
+			var vertical = VerticalAlignment;
+			var stretch = Stretch;
+			var userMinSize = new Size(MinWidth, MinHeight);
+			var userSize = new Size(Width, Height);
+
+			var size = userSize;
+
+			// Like for the measure, if no user size defined on a given axis, we try to stretch along this axis
+			if (double.IsNaN(size.Width))
+			{
+				size.Width = stretch == Stretch.UniformToFill || HorizontalAlignment == HorizontalAlignment.Stretch
+					? finalSize.Width
+					: 0;
+			}
+			if (double.IsNaN(size.Height))
+			{
+				size.Height = stretch == Stretch.UniformToFill || VerticalAlignment == VerticalAlignment.Stretch
+					? finalSize.Height
+					: 0;
+			}
+
+			// Like for the measure, in case userSize was not defined, we still have to apply the min size
+			size = size
+				.AtLeast(userMinSize)
+				.NumberOrDefault(userMinSize);
+
+			// The area that will be used to render the rectangle/ellipse as path
+			var pathArea = new Rect(new Point(), size);
+
+			// Apply the stretch mode, as it might change the "shape" of a "relative shape"
+			switch (stretch)
+			{
+				case Stretch.None:
+					pathArea.Height = pathArea.Width = 0;
+					break;
+
+				default:
+				case Stretch.Fill:
+					// size is already valid ... nothing to do!
+					break;
+
+				case Stretch.Uniform when pathArea.Width < pathArea.Height:
+					pathArea.Height = pathArea.Width;
+					break;
+
+				case Stretch.Uniform: // when pathArea.Width > pathArea.Height:
```

Comment:
Comment not needed ?

Reply:
just to explicit the case ... not really needed indeed